drivers/storage/portworx/deployment.go

```diff
@@ -361,9 +361,43 @@ func (p *portworx) GetStoragePodSpec(
 		}
 	}
 
+	p.pruneVolumes(&podSpec)
+
 	return podSpec, nil
 }
 
+func (p *portworx) pruneVolumes(spec *v1.PodSpec) {
+	seen := make(map[string]bool)
+	for ci, container := range spec.Containers {
+		seenDest := make(map[string]bool)
+		vm := container.VolumeMounts
+		for i := len(vm) - 1; i >= 0; i-- {
+			if seenDest[vm[i].MountPath] {
```

Comment:
you can use path.Clean function to convert the paths, which handles extra slashes in the end, ../, ./ etc in the path that may cause same paths to have different string

Reply:
I am aware of the `path.Clean()` function -- but this is not what we are doing here:
* we will scan the volume mounts in the containers, and record all mounts into `seen`, and all mount destinations into `seenDest`
* if we find any mounts destination repeated (e.g. `-v /dir1:/dest -v /dir2:/dest ...`), we will simply exclude the first mount, and "let the last mount win"

Reply:
I got that, with string comparison mounts destination /dest and /dest/ (with extra slash) will be taken as different destination.

Reply:
Oh, I see your point -- the `/dest`, `/dest/` and `/xx/../dest` would all be treated as a different mount directory; we should `path.Clean()` the paths before comparison, so they evaluate to the same directory,

Reply:
yes

Reply:
see 3953cb8